docs/src/Run/Scalability.md

```diff
@@ -1,3 +1,169 @@
+    - [HPA Not Scaling Pods](#hpa-not-scaling-pods)
+    - [Scaling Too Slowly or Too Aggressively](#scaling-too-slowly-or-too-aggressively)
+
+---
+
+## **Introduction**
+
+Kubernetes Horizontal Pod Autoscaler (HPA) helps scale your deployment based on real-time resource usage (such as CPU and memory). By configuring HPA for the Frequency Gateway, you ensure your services remain available and responsive under varying loads, scaling out when demand increases and scaling down when resources aren't needed.
+
+---
+
+## **Prerequisites**
+
+Before implementing autoscaling, ensure that:
+
+- Kubernetes metrics server is enabled and running (or another resource metrics provider).
+- [**Helm**](https://helm.sh/docs/intro/install/) installed for managing Kubernetes applications.
+- The deployment for [**Frequency Gateway**](https://github.com/ProjectLibertyLabs/gateway/blob/main/deployment/k8s) is running in your Kubernetes cluster.
+
+---
+
+## **Configuring Horizontal Pod Autoscaler**
+
+### **Default Autoscaling Settings**
+
+In `values.yaml`, autoscaling is controlled with the following parameters:
+
+```yaml
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 5
+  targetCPUUtilizationPercentage: 80
+  targetMemoryUtilizationPercentage: 70
+```
+
+- **enabled**: Enable or disable autoscaling.
+- **minReplicas**: Minimum number of pod replicas.
+- **maxReplicas**: Maximum number of pod replicas.
+- **targetCPUUtilizationPercentage**: Average CPU utilization target for triggering scaling.
+- **targetMemoryUtilizationPercentage**: Average memory utilization target for triggering scaling.
+
+---
+
+### **Metrics for Autoscaling**
+
+The Kubernetes HPA uses real-time resource consumption to determine whether to increase or decrease the number of pods. Metrics commonly used include:
+
+- **CPU utilization**: Scaling based on CPU usage.
+- **Memory utilization**: Scaling based on memory consumption.
+
+You can configure one or both, depending on your resource needs.
+
+---
+
+### **Sample Configuration**
+
+Here is an example `values.yaml` configuration for enabling autoscaling with CPU and memory targets:
+
+```yaml
+autoscaling:
+  enabled: true
+  minReplicas: 2
+  maxReplicas: 10
+  targetCPUUtilizationPercentage: 70
+  targetMemoryUtilizationPercentage: 75
+```
+
+This setup will ensure the following:
+
+- The number of pod replicas will never go below 2 or above 10.
+- Kubernetes will attempt to keep CPU usage around 70% across all pods.
+- Kubernetes will attempt to keep memory usage around 75% across all pods.
+
+---
+
+## **Resource Limits**
+
+Setting resource limits ensures your pods are scheduled appropriately and have the necessary resources to function efficiently. Define limits and requests in the `values.yaml` like this:
+
+```yaml
+resources:
+  requests:
+    memory: "256Mi"
+    cpu: "100m"
+  limits:
+    memory: "512Mi"
+    cpu: "500m"
+```
+
+- **requests**: The amount of CPU and memory a pod is guaranteed.
```

Comment:
I am not sure what this means.  The amount of CPU and memory in a pod is guaranteed?  Even if a pod can have more, the following amount is guaranteed to be available?

Reply:
Yes the minimum requested is always available.

Reply:
Improved the wordings here for clarity